File: atividade-4/032_sequencia/032_sequencia.js
//Gerar Lista [m, m+1, m+2, ..., m+n-1]

const n1 = 3;
const n2 = 5;

const sequencia = (x, y) => {
  if (!x) return [];

  return [y].concat(sequencia(x - 1, y + 1));
};

console.log(sequencia(n1, n2));

File: atividade-4/032_sequencia/032_sequencia.hs
module Main where

-- Gerar Lista [m, m+1, m+2, ..., m+n-1]

sequencia :: Int -> Int -> [Int]
sequencia 0 x = []
sequencia x y = y : sequencia (x - 1) (y + 1)

main = do
  a <- readLn :: IO Int
  b <- readLn :: IO Int
  print $ sequencia a b